solidity/contracts/bank/Bank.sol

```diff
@@ -254,6 +257,31 @@ contract Bank is Ownable {
         _increaseBalance(recipient, amount);
     }
 
+    /// @notice Increases the given smart contract `vault`'s balance and
+    ///         notifies the `vault` contract. Called by the Bridge after
+    ///         the deposits routed by depositors to that `vault` have been
+    ///         swept by the Bridge. This way, the depositor does not have to
+    ///         issue a separate  transaction to the `vault` contract.
+    ///         Can be called only by the Bridge.
+    /// @dev The `vault` must implement `IVault` intrface.
+    /// @param vault Address of `IVault` recipient contract
+    /// @param depositors Addresses of depositors whose deposits have been swept
+    /// @param depositedAmounts Amounts deposited by individual depositors and
+    ///        swept. The `vault`'s balance in the Bank will be increased by the
+    ///        sum of all elements in this array.
+    function increaseBalanceAndCall(
```

Comment:
The `vault` can be theoretically untrusted since the depositor can pass an arbitrary value. A malicious vault can implement `onBalanceIncreased` which can reentry `sweep` again or make the entire transaction out of gas. Shouldn't we protect against that situation by adding vaults whitelisting in the Bridge and/or reentrancy guards where applicable?

Reply:
Correct. This is covered in https://github.com/keep-network/tbtc-v2/issues/95. I think only the approved vaults should be able to use the shortcut. Everyone is welcome to implement their own vault but they need to use the standard path: increase bank balance, approve and call vault (or do the last two in one TX, see https://github.com/keep-network/tbtc-v2/issues/99). I was initially planning to implement it in this PR but did now want to cause conflicts with changes in the Bridge from #90.

An alternative approach could be to let `IVault.onBalanceIncreased` fail and execute it with a gas limit. But I think this path is way riskier and may lead to bank balances being locked forever under some vault without proper logic executed (e.g. balance transferred but tokens not minted).

Reply:
Ok, wasn't aware of #95. Thanks for explanation!